adapters/axonix/axonix.go

```diff
@@ -0,0 +1,135 @@
+package axonix
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	URI string
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &adapter{
+		URI: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No impressions in the bid request",
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	requestJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	errors := make([]error, 0, 1)
+
+	var bidderExt adapters.ExtImpBidder
+	err = json.Unmarshal(request.Imp[0].Ext, &bidderExt)
+
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	var axonixExt openrtb_ext.ExtImpAxonix
+	err = json.Unmarshal(bidderExt.Bidder, &axonixExt)
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: "ext.bidder.supplyId not provided",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if axonixExt.SupplyId == "" {
+		err = &errortypes.BadInput{
+			Message: "supplyId is empty",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	thisURI := a.URI
+	if len(thisURI) == 0 {
+		thisURI = "https://openrtb-us-east-1.axonix.com/supply/prebid-server/" + axonixExt.SupplyId
```

Comment:
`SupplyId` is a UUID that will be assigned to each supply partner so highly unlikely if not impossible that it will come with any other spurious character.